### ISSUE TITLE
bug(dialog): Chromevox doesn't announce dialog role

### ISSUE THREAD

**Issue by calebegg** [angular/components#21840]
Compare https://material.angular.io/components/dialog/overview vs https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html

The former doesn't say it's a dialog when you enter it. The latter does.

**Comment by crisbeto** [comment]
I spent a bit of time looking into it and the problem seems to be due to the fact that we move focus to the dialog container until the content is rendered and we can move it to the first focusable element. The reason we move focus to the dialog is so that the user doesn't accidentally open multiple dialogs while the current dialog is animating. An alternate approach could be to `blur` the the focused element while the dialog is animating, but that might not be ideal for accessibility either.

**Comment by jelbourn** [comment]
Possibly the same thing as #10591

This came up during our most recent round of testing. It looks like it only occurs when
* The dialog is not configured to focus the dialog element on open
* Animations are enabled

**Comment by Splaktar** [comment]
I can reproduce this with the MDC-dialog in the dev-app. When the dialog is opened, either the focused input is read ("How much? Edit text") or the focused button is read ("Close Button Press Search plus Space to activate"). In both cases, no indication is given that a modal dialog has been opened and that focus has been moved inside of that modal dialog.

If you press Search+Up Arrow enough times, you will eventually get to the dialog and get "Neptune Dialog" read out. So it's there, but as Kristiyan mentioned, the rapid swap of the focus to the dialog container and then to the first focusable element causes the screen reader to bail out and announce nothing.
